TestCases/compliance-level-3/0104-allowed-values/0104-allowed-values.dmn

```diff
@@ -0,0 +1,1092 @@
+
+    <!-- a context type with a constrained list property -->
+    <dmn:itemDefinition name="tFooBarListContext">
+        <dmn:itemComponent name="prop1">
+            <dmn:typeRef>tFooBarList</dmn:typeRef>
+        </dmn:itemComponent>
+    </dmn:itemDefinition>
+
+    <!-- a type suitable for a relation - a list of contexts, though
+    each with a contrained property -->
+    <dmn:itemDefinition name="tFooBarRelation" isCollection="true">
+        <dmn:itemComponent name="col1">
+            <dmn:typeRef>tFooBarString</dmn:typeRef>
+        </dmn:itemComponent>
+    </dmn:itemDefinition>
+
+    <!-- a list with a contrained string type -->
+    <dmn:itemDefinition name="tFooBarList" isCollection="true">
+        <dmn:typeRef>tFooBarString</dmn:typeRef>
+    </dmn:itemDefinition>
+
+    <dmn:itemDefinition name="tFooBarString">
+        <dmn:typeRef>string</dmn:typeRef>
+        <dmn:allowedValues>
+            <dmn:text>"foo", "bar"</dmn:text>
+        </dmn:allowedValues>
+    </dmn:itemDefinition>
+
+    <dmn:itemDefinition name="tFooBarString_copy">
+        <!-- note we're using the tFooBarString as a base type here and not providing
+        allowedValue.  We expect the allowed values of tFooBarString to pertain to
+        this type as well -->
+        <dmn:typeRef>tFooBarString</dmn:typeRef>
+    </dmn:itemDefinition>
+
+    <!-- function type with constrained return type -->
+    <dmn:itemDefinition name="tFooBarFunction">
+        <dmn:functionItem outputTypeRef="tFooBarString">
+            <dmn:parameters name="a"/>
+        </dmn:functionItem>
+    </dmn:itemDefinition>
+
+    <!-- function type for decision service with untyped param and constrained return type -->
+    <dmn:itemDefinition name="tFooBarDSFunctionWithTypedReturn">
+        <dmn:functionItem outputTypeRef="tFooBarString">
+            <dmn:parameters name="tUntypedInput"/>
+        </dmn:functionItem>
+    </dmn:itemDefinition>
+
+    <!-- function type for decision service with constrained param and no return type -->
+    <dmn:itemDefinition name="tFooBarDSFunctionWithTypedParam">
+        <dmn:functionItem>
+            <dmn:parameters name="ds_inputdecision" typeRef="tFooBarString"/>
+        </dmn:functionItem>
+    </dmn:itemDefinition>
+
+    <!-- function type with a constrained parameter type -->
+    <dmn:itemDefinition name="tTypedParamFooBarFunction">
+        <dmn:functionItem>
+            <dmn:parameters name="a" typeRef="tFooBarString"/>
+        </dmn:functionItem>
+    </dmn:itemDefinition>
+
+    <dmn:itemDefinition name="tDirectionsString">
+        <!-- note this type uses another string type as a base.  We expect
+        the allowed values to be 'overridden' here.  That is, the allowed values
+        in tFooBarString will not longer apply -->
+        <dmn:typeRef>tFooBarString</dmn:typeRef>
+        <dmn:allowedValues>
+            <dmn:text>"left", "right"</dmn:text>
+        </dmn:allowedValues>
+    </dmn:itemDefinition>
```

Comment:
TCK 2021-07-15
the group is not ootb agreeing that "allowedValues" do any overridden behaviour, instead is restriction on restricitons on the extension hierarchy chain.

But we also need to check in the DMN Specification if there are any explicit mandates about the checking for allowedValues. And potentially something about the XML spec?

Reply:
(we will also raise this to the RTF group)

Reply:
(there could be the need to add something also on the chapter of Equivalence and Conformance)

Reply:
Hi @tarilabs @opatrascoiu  - yes, the spec is not very helpful here.  The whole notion of allowedValues is fairly vague. Precisely why I did this suite. 

If it helps, I was going on this in "7.3.3 ItemDefinition metamodel"

> If an ItemDefinition element contains one or more allowedValues, the allowedValues specifies the complete range of values that this ItemDefinition represents.

No notions of extension or hierarchy.